source/solvers/navier_stokes_base.cc

```diff
@@ -876,19 +876,23 @@ NavierStokesBase<dim, VectorType, DofsType>::postprocess(bool firstIter)
         }
     }
 
-  // Time-averaged velocities are calculated if calculate average velocities
-  // and/or calculate reynolds stress is enabled because the last one needs
-  // the time-averaged values at each time step.
   // The average_solution and reynolds_stress vectors are reinitialize at the
   // first iteration.
-  // The average velocities and/or reynolds stresses are calculated when the
+  // The average velocities and reynolds stresses are calculated when the
   // time reaches the initial time. (time >= initial time) with 1e-6 as
   // tolerance.
-  if (this->nsparam.post_processing.calculate_average_velocities ||
-      this->nsparam.post_processing.calculate_reynolds_stress)
+  if (this->nsparam.post_processing.calculate_average_velocities)
     {
       if (firstIter)
         {
+          if (nsparam.mesh_adaptation.type ==
+              Parameters::MeshAdaptation::Type::none)
+            AssertThrow(true,
+                        ExcMessage(
+                          "Time-averaging velocities and calculating reynolds "
+                          "stresses are currently unavailable for mesh "
+                          "adaptation."));
+
```

Comment:
This is not correct. You want to assert throw nsparam.mesh_adaptation.type == none.
What this means is AssertThrow(nsparam.mesh_adaptation.type==none,...) If the condition is not true, it will throw and error and stop execution of the program. Right now this is working because you are throwing true, but true will always be true and thus always pass.

Reply:
Ohhhhhh yeah I get it!